### PR TITLE
feat: changes to forms now that title code no longer mandatory

### LINKED ISSUE #872: Changes to forms now that title code no longer mandatory
Title code no longer required (dependency on RAY-74), we'll remove it on some pages and make sure it is optional for others

Remove from:
- Registration page
- Shipping and billing address pages during checkout

Display but make sure it is optional from:
- Address management edit/add (add "None" option but send blank to backend; make "None" the default in grey instead of "Select title")

Check if elsewhere.

Check related tickets #440 and #287 and see if they are fixed or we can close them. If not, pull them into the sprint and fix them.

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #872

### CODE REVIEW

File: projects/core/src/user/facade/user.service.ts [hunk 1]
```diff
@@ -37,28 +38,10 @@ export class UserService {
...
-    this.store.dispatch(
-      new fromStore.RegisterUser({
-        firstName: firstName,
-        lastName: lastName,
-        password: password,
-        titleCode: titleCode,
-        uid: email
-      })
-    );
+  register(submitFormData: UserRegisterFormData) {
```
Comment (kacperknapik) [inline projects/core/src/user/facade/user.service.ts]: `submitFormData` is imho badly named. let's change it to something more semantic

File: projects/core/src/user/model/user.model.ts [hunk 2]
```diff
@@ -3,5 +3,4 @@ export interface UserRegisterFormData {
   firstName: string;
   uid: string;
   password: string;
-  titleCode: string;
```
Comment (kacperknapik) [inline projects/core/src/user/model/user.model.ts]: shouldn't we leave it here, but as a optional prop? @Xymmer

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 3]
```diff
@@ -14,7 +14,6 @@
                   [items]="titles"
                   bindLabel="name"
                   bindValue="code"
-                  placeholder="Select Title..."
```
Comment (kacperknapik) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]: why exactly was this placeholder attribute deleted?

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.spec.ts [hunk 4]
```diff
@@ -181,7 +182,7 @@ describe('AddressFormComponent', () => {
       .unsubscribe();
 
     expect(countries).toBe(mockCountries);
-    expect(titles).toBe(mockTitles);
+    expect(titles).toEqual([{ code: '', name: 'None' }, ...mockTitles]);
```
Comment (kacperknapik) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.spec.ts]: you should rather add this 'none' titleCode to the mockTitles array, am I right?

File: projects/storefrontlib/src/lib/user/register/register.component.ts [hunk 5]
```diff
@@ -86,13 +76,13 @@ export class RegisterComponent implements OnInit, OnDestroy {
...
 
   submit(): void {
-    this.userService.register(
-      this.userRegistrationForm.value.titleCode,
-      this.userRegistrationForm.value.firstName,
-      this.userRegistrationForm.value.lastName,
-      this.userRegistrationForm.value.email,
-      this.userRegistrationForm.value.password
-    );
+    const submitFormData: UserRegisterFormData = {
```
Comment (kacperknapik) [inline projects/storefrontlib/src/lib/user/register/register.component.ts]: once again `submitFormData` is not semantic enough

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.spec.ts (hunk 4 above)
Comment (Pucek9) [reply projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.spec.ts]: not really, then 'none' option will be duplicated. But I can separate it to another array

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html (hunk 3 above)
Comment (Pucek9) [reply projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]: After add option "none", this select is always selected so placeholder will be not displayed anymore

File: projects/core/src/user/model/user.model.ts (hunk 2 above)
Comment (Pucek9) [reply projects/core/src/user/model/user.model.ts]: in the first approach we had to remove it from registration at all. Now its reverted

**Comment by Matejk00** [comment]
Please resolve all conflicts.

**Pull request merged**